src/ray/raylet/worker_pool.cc

```diff
@@ -723,7 +728,9 @@ std::shared_ptr<WorkerInterface> WorkerPool::PopWorker(
 
   std::shared_ptr<WorkerInterface> worker = nullptr;
   Process proc;
-  if (task_spec.IsActorCreationTask() && !task_spec.DynamicWorkerOptions().empty()) {
+  if (task_spec.IsActorCreationTask() && (!task_spec.DynamicWorkerOptions().empty() ||
+                                          (RayConfig::instance().enable_multi_tenancy() &&
```

Comment:
I don't think multi-tenancy is required here since we already support custom JVM options (dynamic worker options) for Java actors with multi-tenancy off.

Reply:
When there are both options provided, per-worker env overrides the job config?